src/abstract-project-create-command.ts

```diff
@@ -140,69 +119,119 @@ export default abstract class AbstractProjectCreateCommand extends AbstractAuthe
     assertPresent({
       authenticator,
       eventSender,
-      commandGenerateConfigGetter,
       projectCreator,
       database,
       messages,
       spinner,
     });
 
     this.eventSender = eventSender;
-    this.commandGenerateConfigGetter = commandGenerateConfigGetter;
     this.projectCreator = projectCreator;
     this.database = database;
     this.messages = messages;
     this.spinner = spinner;
   }
 
+  protected async runAuthenticated(): Promise<void> {
+    try {
+      const { appConfig, dbConfig, meta, authenticationToken } = await this.getConfig();
+
+      this.spinner.start({ text: 'Creating your project on Forest Admin' });
+      const projectCreationPromise = this.projectCreator.create(
+        authenticationToken,
+        appConfig,
+        meta,
+      );
+      const { id, envSecret, authSecret } = await this.spinner.attachToPromise(
+        projectCreationPromise,
+      );
+
+      this.eventSender.meta.projectId = id;
+
+      await this.testDatabaseConnection(dbConfig);
+
+      await this.generateProject({
+        dbConfig,
+        appConfig,
+        forestAuthSecret: authSecret as string,
+        forestEnvSecret: envSecret as string,
+      });
+
+      await this.notifySuccess();
+    } catch (error) {
+      // Display customized error for non-authentication errors.
+      if (error.status !== 401 && error.status !== 403) {
```

Comment:
Knowing we are in a `runAuthenticated` method, checking if the error is a 401 one is incorrect to me. At this stage, we are absolutely that the user is authenticated. So only 403 (authorization error) should and can occure here.

Reply:
The token of the user can be discarded during the command execution.
I simply moved this code around, this is not new code.

Reply:
👌